### PR TITLE
feat(ui): Display the government name on the planet card

### LINKED ISSUE #4022: Show government name of a planet when different from system's government
**Describe the bug**
When a planet's government is different from the system's government, the map screen doesn't indicate that.

**To Reproduce**
1. Go to the map screen
2. Click on Tarazed
3. Observe that there's no indication that Echo belongs to the Quarg

**Expected behavior**
The indication would be useful, especially if events changed a planet's government while the player is in another system.

**Screenshots**
screenshot_2018-11-21_04-26-16

**System (please complete the following information):**
Current master

**Additional context**
Was previously submitted as PR #3768. That PR was closed for UI reasons, but the problem hasn't been fixed, so I'm resubmitting (with @Pointedstick 's blessing) in the hope that another solution will be found.

### PR DESCRIPTION
**Pull request by TomGoodIdea**
**Feature:** This PR resolves #4022
## Feature Details
The name of the planet's government is now displayed on the planet card (map UI), above the friendliness line. Selecting the entry activates the government map mode.

This is enabled for systems that have a planet with government different from that of the system (planets with the special "Uninhabited" government aren't considered).

## UI Screenshots
<details>
<summary>Before</summary>

b1

b2
</details>

<details>
<summary>After</summary>

a1
(no difference here)

tarazed
</details>

## Testing Done
Displays correctly, selecting works.

## Performance Impact
N/A

### CODE REVIEW

**Comment by Hurleveur** [comment]
Looks good code wise but we could also only do this if the governments differ in the system?
I mean it's not like we are limited on space, it is endless, but for small screens adding a line means less shown

File: data/_ui/interfaces.txt [hunk 1]
```diff
@@ -185,10 +185,10 @@ interface "map detail panel"
 interface "map planet card"
 	value "text start" 13
 	value "category size" 20
-	value "height" 130
+	value "height" 150
```
Comment (Hurleveur) [inline data/_ui/interfaces.txt:188]: just gets me thinking, since we have category size we should probably just have this be named "extra height" and in this case be 30? (150 - 6 * 20, since category size is already something we have)
we can also just name it margin

you dont have to do this though I could do it in another PR or submit a PR to this one

Comment (TomGoodIdea) [reply data/_ui/interfaces.txt:188]: I prefer "extra height"

**Comment by TomGoodIdea** [comment]
What do you think about the `Uninhabited` government? Should it be displayed with the `medium` color, or the `faint` color?

File: data/_ui/interfaces.txt (hunk 1 above)
Comment (Hurleveur) [reply data/_ui/interfaces.txt:188]: Do you need me to do it?
By the way just curious was this code easy enough to modify? It is a bit complex but I tried to make it easily modifiable

Comment (TomGoodIdea) [reply data/_ui/interfaces.txt:188]: It wasn't hard to show an additional row on all cards, but now we need to support variable card height, which requires a MapPlanetCard / MapDetailPanel rework, to make things like available space calculations don't assume fixed height.
If you could do that it would be really helpful.

**Review by RisingLeaf — APPROVED** [review]
looks good and simple

File: data/_ui/interfaces.txt (hunk 1 above)
Comment (Hurleveur) [reply data/_ui/interfaces.txt:188]: I think this needs a bit more thought put into it for how it'd be best to code it
The easiest way is probably to just figure out if we have different govs in the system and if we do display the gov of all planets? idk

**Comment by Hurleveur** [comment]
@TomGoodIdea I updated my PR to your branch to fix bugs it used to have and allow to have it only be displayed when the govs differ

**Review by Hurleveur — APPROVED** [review]
Since I tested it and coded a part of it I naturally approve of this

File: source/MapPlanetCard.cpp [hunk 2]
```diff
@@ -43,6 +47,11 @@ MapPlanetCard::MapPlanetCard(const StellarObject &object, unsigned number, bool
 	hasSpaceport = planet->HasSpaceport();
 	hasShipyard = planet->HasShipyard();
 	hasOutfitter = planet->HasOutfitter();
+	governmentName = planet->GetGovernment()->GetName();
+	if(systemGovernmentName.empty())
+		systemGovernmentName = planet->GetSystem()->GetGovernment()->GetName();
+	if(planet->GetGovernment()->GetName() != "Uninhabited" && governmentName != systemGovernmentName)
+		hasGovernments = true;
```
Comment (warp-core) [inline source/MapPlanetCard.cpp:54]:
```suggestion
	governmentName = planet->GetGovernment()->GetName();
	if(systemGovernmentName.empty())
		systemGovernmentName = planet->GetSystem()->GetGovernment()->GetName();
	if(governmentName != "Uninhabited" && governmentName != systemGovernmentName)
		hasGovernments = true;
```
Isn't this what you just assigned to this variable?

File: source/MapPlanetCard.h [hunk 3]
```diff
@@ -86,10 +90,14 @@ class MapPlanetCard {
...
 	float spriteScale;
 
+	std::string governmentName;
 	std::string reputationLabel;
 	const std::string &planetName;
 	// The currently select category (outfitter, shipyard, ...)
 	unsigned selectedCategory = 0;
+
+	static std::string systemGovernmentName;
+	static bool hasGovernments;
```
Comment (warp-core) [inline source/MapPlanetCard.h:100]: Why are these static?
If they're going to stay static, they should probably just go in an anonymous namespace in MapPlanetCard.cpp.
But, I don't think they should be static at all.
`systemGovernmentName` doesn't even seem to need to be a class member at all. It is only ever set or accessed in the constructor, apart from the `ResetSize` static method, which just clears it, which wouldn't be something that needs to be done if it weren't a static class member but instead a local in the constructor.

Comment (Hurleveur) [reply source/MapPlanetCard.h:100]: agreed, do you see how to move it to the namespace @TomGoodIdea ?

Comment (Hurleveur) [reply source/MapPlanetCard.h:100]: It has to be a static because it must be the same for all planetcards: either they all disply it or not

**Comment by warp-core** [comment]
I don't think the governent should be presented in the same style as the other three lines there. The spaceport, shipyard, and outfitter are services, and those lines are providing a different type of information to the name of the government.
Maybe those lines could be indented slightly further?
Or, the government coul be listed closer to the name of the planet like system governments are listed below system names?

**Comment by Hurleveur** [comment]
If we indent those lines further we would need to make the panel larger ~~I dont want to give it changing width too~~ and since this is something that happens say in what 4 systems of the game basic support for it sounds fine to me tbh
Has been visited is not really a service it's just an information, all of which have legends, like government does now

**Review by OcelotWalrus — APPROVED** [review]
I looks good and it seems logic to add this.

**Comment by lumbar527** [comment]
What if it was like this:
```
Wayfarer (Free Worlds)
    Friendly
    Shipyard
    Outfitter
    (Has been visited)
```

**Comment by OcelotWalrus** [comment]
> What if it was like this?
> 
> ```
> Wayfarer (Free Worlds)
>     Friendly
>     Shipyard
>     Outfitter
>     (Has been visited)
> ```
Ooh! It seems to be better like that.

File: source/MapPlanetCard.cpp [hunk 4]
```diff
@@ -182,23 +193,26 @@ bool MapPlanetCard::DrawIfFits(const Point &uiPoint)
 		};
 
 		// Draw the name of the planet.
-		if(FitsCategory(5.))
+		if(FitsCategory(categories + hasGovernments))
 			font.Draw({ planetName, alignLeft }, uiPoint + Point(0, textStart), isSelected ? medium : dim);
```
Comment (OcelotWalrus) [inline source/MapPlanetCard.cpp:197]:
```suggestion
			font.Draw({ planetNameFull, alignLeft }, uiPoint + Point(0, textStart), isSelected ? medium : dim);
```

File: source/MapPlanetCard.cpp [hunk 5]
```diff
@@ -182,23 +193,26 @@ bool MapPlanetCard::DrawIfFits(const Point &uiPoint)
 		};
 
 		// Draw the name of the planet.
```
Comment (OcelotWalrus) [inline source/MapPlanetCard.cpp:195]:
```suggestion
		// Draw the name of the planet.
		string planetNameFull = planetName + " (" + governmentName + ")";
```

**Review by OcelotWalrus — CHANGES_REQUESTED** [review]
The planet name and the government name should be concentrated.
Instead of:
```
Planet
  Government
```
it will be
```
Planet (Government)
```

**Comment by Hurleveur** [comment]
I dont think that's a good idea because it will easily be too small for big gov/planet names.

**Review by OcelotWalrus — APPROVED** [review]
> I dont think that's a good idea because it will easily be too small for big gov/planet names.

Sure! It is already good how it is right now!

**Comment by TomGoodIdea** [comment]
Could someone please tell me what I am supposed to do with this PR?

**Comment by Amazinite** [comment]
Per @warp-core's comment, change how the government name is displayed so that it's not displayed like a service. The suggestion from Ocelot to put the government next to the name definitely doesn't have enough space, but warp suggested possibly making it look closer to how the planet system and government names look in the UI below the planet cards. We could perhaps also make it closer to the visited/not visited line.
image

File: source/MapPlanetCard.h [hunk 6]
```diff
@@ -60,6 +61,9 @@ class MapPlanetCard {
 
 	void Select(bool select = true);
 
+	static double Height();
+
+	static void ResetSize();
```
Comment (Amazinite) [inline source/MapPlanetCard.h:67]: Maintain two lines of whitespace between the last function in `public:` and `protected:`
```suggestion


```

File: data/_ui/interfaces.txt [hunk 7]
```diff
@@ -185,10 +185,10 @@ interface "map detail panel"
 interface "map planet card"
 	value "text start" 13
 	value "category size" 20
-	value "height" 130
+	value "categories" 5
+	value "extra height" 30
```
Comment (Amazinite) [inline data/_ui/interfaces.txt:189]: Might "height padding" by a more descriptive name?

Comment (Hurleveur) [reply data/_ui/interfaces.txt:189]: Yeah that does sound better
